KLSadd_insertion/src/updateChapters.py

```diff
@@ -187,162 +468,197 @@ def referencePlacer(chap, references, p, kls, refII, refIII, specref):
-    # we can use the section names to place the relevant paragraphs in the
-    # right place
-
-    # as of 2/8/16 the paragraphs will go before the References paragraph of the relevant section
-    # parse both files 9 and 14 as strings
-
-    # chapter 9
-    with open("chap09.tex", "r") as ch9:
-        entire9 = ch9.readlines()  # reads in as a list of strings
-    ch9.close()
-
-    # chapter 14
-    with open("chap14.tex", "r") as ch14:
-        entire14 = ch14.readlines()
-    ch14.close()
-    # call the findReferences method to find the index of the References paragraph in the two file strings
-    # two variables for the references lists one for chapter 9 one for chapter
-    # 14
-    chapticker = 0
-    references9 = findReferences(entire9)
-    chapticker += 1
-    references14 = findReferences(entire14)
-    # call the fixChapter method to get a list with the addendum paragraphs
-    # added in
-    chapticker2 = 0
-    str9 = ''.join(
-        fixChapter(
-            entire9,
-            references9,
-            paras,
-            addendum,
-            ref9II,
-            ref9III,
-            specref9))
-    chapticker2 += 1
-    str14 = ''.join(
-        fixChapter(
-            entire14,
-            references14,
-            paras,
-            addendum,
-            ref14II,
-            ref14III,
-            specref14))
-
-"""
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-If you are writing something that will make a change to the chapter files, write it BEFORE this line, this part
-is where the lists representing the words/strings in the chapter are joined together and updated as a string!
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""
-
-
-# write to files
-# new output files for safety
-with open("updated9.tex", "w") as temp9:
-    temp9.write(str9)
-
-with open("updated14.tex", "w") as temp14:
-    temp14.write(str14)
+
+def main():
+    """
+    Runs all of the other functions and outputs their results into an output file as well as putting together the list
+    of additions that is fed into fix_chapter.
+
+    :return:
+    """
+    # open the KLSadd file to do things with
+    with open(DATA_DIR + "KLSaddII.tex", "r") as add:
+        # store the file as a string
+        addendum = add.readlines()
+        kls_list_all = new_keywords(addendum)
+        # Makes sections look like other sections
+
+        for word in addendum:
+            if "paragraph{" in word:
+                lenword = len(word) - 1
+                temp = word[0:word.find("{") + 1] + "\large\\bf KLSadd: " + word[word.find("{") + 1: lenword]
+                addendum[addendum.index(word)] = temp
+            if "subsubsection*{" in word:
+                lenword = len(word) - 1
+                addendum[addendum.index(word)] = word[0:word.find("{") + 1] + "\large\\bf KLSadd: " + \
+                word[word.find("{") + 1: lenword]
```

Comment:
Hell, this variable is actually useless: you can just do `word[word.find("{"):-1]`.